blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -137,10 +138,13 @@ class RunDetails extends Component {
     afterClose = () => {
         const { router, params } = this.context;
         if (this.opener) {
-            router.goBack();
+            // step back in the history to the item that's prior to initial load of RunDetails
+            const offsetToInitialRoute = this.initialHistoryLength - history.length;
+            router.go(offsetToInitialRoute - 1);
```

Comment:
ok, so maybe we need this. a prior app i worked on had a 'navigation stack', maybe we maintain that separately from the router, we could easily go back in these sorts of cases. it actually worked well, allowed for breadcrumbs and such